AnimationSystem/AnimationSystem.h

```diff
@@ -0,0 +1,137 @@
+    LoopOnce = 2200,
+    LoopRepeat = 2201,
+    LoopPingPong = 2202
+  };
+  enum BoneIndex {
+    // Hips = 0, // position
+    Hips = 1, // quaternion
+    Spine = 2,
+    Chest = 3,
+    UpperChest = 4,
+    Neck = 5,
+    Head = 6,
+    Left_shoulder = 7,
+    Left_arm = 8,
+    Left_elbow = 9,
+    Left_wrist = 10,
+    Left_middleFinger1 = 11,
+    Left_middleFinger2 = 12,
+    Left_middleFinger3 = 13,
+    Left_thumb0 = 14,
+    Left_thumb1 = 15,
+    Left_thumb2 = 16,
+    Left_indexFinger1 = 17,
+    Left_indexFinger2 = 18,
+    Left_indexFinger3 = 19,
+    Left_ringFinger1 = 20,
+    Left_ringFinger2 = 21,
+    Left_ringFinger3 = 22,
+    Left_littleFinger1 = 23,
+    Left_littleFinger2 = 24,
+    Left_littleFinger3 = 25,
+    Right_shoulder = 26,
+    Right_arm = 27,
+    Right_elbow = 28,
+    Right_wrist = 29,
+    Right_middleFinger1 = 30,
+    Right_middleFinger2 = 31,
+    Right_middleFinger3 = 32,
+    Right_thumb0 = 33,
+    Right_thumb1 = 34,
+    Right_thumb2 = 35,
+    Right_indexFinger1 = 36,
+    Right_indexFinger2 = 37,
+    Right_indexFinger3 = 38,
+    Right_ringFinger1 = 39,
+    Right_ringFinger2 = 40,
+    Right_ringFinger3 = 41,
+    Right_littleFinger1 = 42,
+    Right_littleFinger2 = 43,
+    Right_littleFinger3 = 44,
+    Right_leg = 45,
+    Right_knee = 46,
+    Right_ankle = 47,
+    Right_toe = 48,
+    Left_leg = 49,
+    Left_knee = 50,
+    Left_ankle = 51,
+    Left_toe = 52
+  };
+
+  struct Interpolant {
+    unsigned int numParameterPositions;
+    float *parameterPositions;
+    float *resultBuffer;
+    unsigned int numSampleValues;
+    float *sampleValues;
+    unsigned int valueSize;
+  };
+  struct Animation {
+    float duration;
+    std::vector<Interpolant> interpolants;
+    unsigned int currentInterpolantIndex = 0;
+    unsigned int index;
+    std::string name;
+  };
+  struct AnimationMapping { // spec
+    float *dst =  (float *)malloc((4) * sizeof(float));
+    bool isPosition;
+    unsigned int index;
+    bool isFirstBone;
+    bool isLastBone;
+    bool isTop;
+    bool isArm;
+  };
+
+  class Avatar {
+  public:
+    AnimationMixer *mixer;
+
+    std::vector<std::string> strings;
+
+    // values
+    std::string activateAnimationName, fallLoopFrom, sitAnimationName, danceAnimationName, emoteAnimationName, useAnimationName, useAnimationComboName, hurtAnimationName, unuseAnimationName, aimAnimationName, holdAnimationName;
+    std::vector<std::string> useAnimationEnvelopeNames;
+    float activateTime, landTime, fallLoopFactor, fallLoopTime, flyTime, doubleJumpTime, jumpTime, narutoRunTime, narutoRunTimeFactor, danceFactor, crouchMaxTime, emoteFactor, lastEmoteTime, idleWalkFactor, useTime, useAnimationEnvelopeLength, hurtTime, unuseTime, aimTime, aimMaxTime, walkRunFactor, crouchFactor, pickUpTime, forwardFactor, backwardFactor, leftFactor, rightFactor, mirrorLeftFactorReverse, mirrorLeftFactor, mirrorRightFactorReverse, mirrorRightFactor, landTimeS, timeSinceLastMoveS;
```

Comment:
Probably best on separate lines for when we make changes.

Reply:
Updated https://github.com/webaverse/app-wasm/pull/75/commits/873179145010e397f0152674bfcc5b7dc9e9f138 .